starknet_py/utils/data_transformer/data_transformer.py

```diff
@@ -344,6 +344,10 @@ def from_python(
                 f"Provided {len(args)} positional arguments, {len(type_by_name)} possible."
             )
 
+        for key, val in named_arguments.items():
+            if key not in type_by_name:
+                raise TypeError(f"Unnecessary named argument {key}={val} provided.")
+
```

Comment:
That will throw only one wrong parameter but could throw all. Maybe go through the entire dict and collect all of them?